src/processor/Package.ts

```diff
@@ -330,4 +331,63 @@ export class Package {
       pullAt(sd.rules, rulesToRemove);
     });
   }
+
+  private removeDateRules(): void {
+    let allDatesMatch = true;
+    let date: string;
+    [...this.profiles, ...this.extensions, ...this.valueSets, ...this.codeSystems].forEach(sd => {
+      for (const rule of sd.rules) {
+        if (!allDatesMatch) break; // If any date hasn't matched the others, stop looking at other rules
+        if (
+          rule instanceof ExportableCaretValueRule &&
+          rule.caretPath === 'date' &&
+          rule.path === ''
+        ) {
+          const dateValue = rule.value as string; // If the rule assigns a date, the value will be a string.
+          if (!date) date = dateValue; // Set the date to match
+          if (date !== dateValue) {
+            allDatesMatch = false;
+          }
+        }
+      }
+    });
+
+    // If all dates are the same, are defined to the second, and are in GMT,
+    // we can assume they were set by the IG Publisher and can be safely removed.
+    const dateTimeRegex = /^([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)(-(0[1-9]|1[0-2])(-(0[1-9]|[1-2][0-9]|3[0-1])(T([01][0-9]|2[0-3]):[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00)))?)?)?$/;
```

Comment:
Please tell me you found this regex somewhere.  If you did, can you put the link in the comment?  If you didn't, then God bless you for writing this regex.

Reply:
Oh yikes there's just no way in the entire world I could write this. It's the dateTime regular expression from FHIR. I'll add a link.